src/test/java/org/wmn4j/io/musicxml/MusicXmlWriterDomTest.java

```diff
@@ -4,21 +4,145 @@
 package org.wmn4j.io.musicxml;
 
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.wmn4j.io.ParsingFailureException;
 import org.wmn4j.notation.TestHelper;
+import org.wmn4j.notation.builders.MeasureBuilder;
+import org.wmn4j.notation.builders.NoteBuilder;
+import org.wmn4j.notation.builders.PartBuilder;
+import org.wmn4j.notation.builders.ScoreBuilder;
+import org.wmn4j.notation.elements.Durations;
+import org.wmn4j.notation.elements.Note;
+import org.wmn4j.notation.elements.Pitch;
 import org.wmn4j.notation.elements.Score;
+import org.wmn4j.notation.iterators.PartWiseScoreIterator;
+import org.wmn4j.notation.iterators.ScoreIterator;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.fail;
 
 class MusicXmlWriterDomTest {
 
-	MusicXmlWriterDomTest() {
+	@TempDir
+	Path temporaryDirectory;
+
+	private static final String MUSICXML_FILE_PATH = "musicxml/";
+
+
+	private Score readMusicXmlTestFile(String testFileName, boolean validate) {
+		final MusicXmlReader reader = new MusicXmlReaderDom(validate);
+		Score score = null;
+		final Path path = Paths.get(TestHelper.TESTFILE_PATH + MUSICXML_FILE_PATH + testFileName);
+
+		try {
+			score = reader.readScore(path);
+		} catch (final IOException | ParsingFailureException e) {
+			fail("Parsing failed with exception " + e);
+		}
+
+		assertNotNull(score);
+		return score;
+	}
+
+	private Score writeScore(Score score) {
+		MusicXmlWriterDom writer = new MusicXmlWriterDom(score);
+		Path file = temporaryDirectory.resolve("file.xml");
+		writer.writeToFile(file.toString());
+
+		//For debugging
+		//printLines(file);
+
+		final MusicXmlReader reader = new MusicXmlReaderDom(true);
+		Score writtenScore = null;
+
+		try {
+			writtenScore = reader.readScore(file);
+		} catch (final IOException | ParsingFailureException e) {
+			fail("Reading score written by MusicXmlWriterDom failed with exception " + e);
+		}
+
+		assertNotNull(writtenScore);
+		return writtenScore;
+	}
+
+	/**
+	 * Easy way to print the contents of a file for debugging
+	 * @param file Path to the file to print
+	 */
+	private void printLines(Path file) {
```

Comment:
This method could be removed because it's not used.

Reply:
I found this method very useful while writing the code and I was planning to continue using it too, but it is essentially for debugging. Would it be better just to leave it out from the commits?

Reply:
It's generally better to leave out code that's not used, because that way the code stays a lot simpler and easier to understand. That's way I would recommend leaving out debugging code.